src/components/ProjectView/ImageList.vue

```diff
@@ -1,41 +1,59 @@
 <script setup>
-    import { ref } from 'vue'
-    const props = defineProps(['projectImageObjects', 'selectedImages', 'data'])
+import { ref, onMounted } from "vue";
+import { useStore } from "vuex";
 
-    // v-data-table setup variables
-    let headers = ref([
-        { title: 'Image Name', align: 'start', sortable: true, key: 'basefile_name'},
-        { title: 'Time', align:'start', sortable: true, key: 'time'},
-        { title: 'Object', align:'start', sortable: true, key: 'object'},
-        { title: 'Image', align:'start', sortable: false, key: 'image'}
-    ])
-    let items = ref(props.data)
-    let itemsPerPage = ref(15)
-    let selected = ref([])
+const props = defineProps(["data"]);
+const store = useStore();
+
+// v-data-table setup variables
+let headers = ref([
+  { title: "Image Name", align: "start", sortable: true, key: "basefile_name" },
+  { title: "Time", align: "start", sortable: true, key: "time" },
+  { title: "Object", align: "start", sortable: true, key: "object" },
+  { title: "Image", align: "start", sortable: false, key: "image" },
+]);
+let items = ref(props.data);
+let itemsPerPage = ref(15);
+
+// --- Selection Logic ---
+let selected = ref([])
+
+// since item-value for v-data-table is set to basefilename we need to convert it back into an object
+// figuring out how to set item-value to the entire object would be great and then this function can be simplified
+function select(selectedImageNames) {
```

Comment:
you can use the `some` method! I don't quite understand the comment (because you've been working on the list view) but the `some` method checks if some elements of an array pass the test provided by the function. I think that's what you're trying to do, if not, I completely misunderstood.

Reply:
Thats a really useful function to know I'll remember that! In this case the data table is just giving me back the basefile names of the selected objects since it uses them as a key to identify which are selected. So before storing them in the store I need to convert them back to their original image objects and not just the name.